rest/route/project.go

```diff
@@ -185,3 +186,133 @@ func (h *versionsGetHandler) Run(ctx context.Context) gimlet.Responder {
 
 	return gimlet.NewJSONResponse(versions)
 }
+
+/*
+ * Creates project (project_ref)
+ * PUT: /rest/v2/projects/
+ * Perm: superUser
+ */
+type projectCreateHandler struct {
+	projectRef *model.APIProjectRef
+
+	sc data.Connector
+}
+
+func makeProjectCreateRoute(sc data.Connector) gimlet.RouteHandler {
+	return &projectCreateHandler{sc: sc}
+}
+
+func (h *projectCreateHandler) Factory() gimlet.RouteHandler {
+	return &projectCreateHandler{sc: h.sc}
+}
+
+func (h *projectCreateHandler) Parse(ctx context.Context, r *http.Request) error {
+	body := util.NewRequestReader(r)
+	defer body.Close()
+
+	if err := util.ReadJSONInto(body, &h.projectRef); err != nil {
+		return errors.Wrap(err, "Cannot parse the content as JSON!")
+	}
+
+	return nil
+}
+
+func (h *projectCreateHandler) Run(ctx context.Context) gimlet.Responder {
+	createdProject, err := h.sc.CreateProject(h.projectRef)
+
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "Cannot create project!"))
+	}
+
+	apiProject := &model.APIProject{}
+	// The method doesn't accept pointers, we have to dereference it first
+	err = apiProject.BuildFromService(*createdProject)
```

Comment:
omit the blank lines between the method calls and the if error lines. 

also modify BuildFromService (or call sites) so you don't have to de-reference pointers at the call site.

Reply:
Isn't it a bad practice to make implicit function signature overloading? It may lead to runtime type casting errors (potentially).